src/uct/cuda/cuda_copy/cuda_copy_ep.c

```diff
@@ -109,34 +114,41 @@ uct_cuda_copy_post_cuda_async_copy(uct_ep_h tl_ep, void *dst, void *src,
     ucs_memory_type_t src_type;
     ucs_memory_type_t dst_type;
     CUstream *stream;
+    CUcontext current_ctx;
     ucs_queue_head_t *event_q;
+    uct_cuda_copy_per_ctx_rsc_t *ctx_rsc;
 
     if (!length) {
         return UCS_OK;
     }
 
+
     /* ensure context is set before creating events/streams */
-    if (iface->cuda_context == NULL) {
-        UCT_CUDADRV_FUNC_LOG_ERR(cuCtxGetCurrent(&iface->cuda_context));
-        if (iface->cuda_context == NULL) {
-            ucs_error("attempt to perform cuda memcpy without active context");
+    UCT_CUDADRV_FUNC_LOG_ERR(cuCtxGetCurrent(&current_ctx));
+    if (current_ctx == NULL) {
+        ucs_error("attempt to perform cuda memcpy without active context");
+        return UCS_ERR_IO_ERROR;
+    } else {
+        status = uct_cuda_copy_get_ctx_rscs(iface, current_ctx, &ctx_rsc);
+        if (UCS_OK != status) {
+            ucs_error("unable to get resources associated with cuda context");
             return UCS_ERR_IO_ERROR;
         }
     }
```

Comment:
This block of code (lines 128-137) is repeated in put_short and get_short as well. Maybe put it in a function or macro?